### PR TITLE
KAFKA-8600: Use RPC generation for DescribeDelegationTokens protocol

### PR DESCRIPTION
**Pull request by viktorsomogyi**
Refactors the DescribeDelegationToken to use the generated RPC classes.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by viktorsomogyi** [comment]
Seems like there's a couple of tests failing, will take a look at them.

**Comment by viktorsomogyi** [comment]
retest this please

**Comment by viktorsomogyi** [comment]
retest this please

**Comment by viktorsomogyi** [comment]
@mimaison @omkreddy would you please review this?

File: core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala [hunk 1]
```diff
@@ -49,6 +47,7 @@ import org.apache.kafka.common.message.RenewDelegationTokenRequestData
 import org.apache.kafka.common.message.SaslAuthenticateRequestData
 import org.apache.kafka.common.message.SaslHandshakeRequestData
 import org.apache.kafka.common.message.SyncGroupRequestData
+import org.apache.kafka.common.message._
```
Comment (mimaison) [inline core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala:30]: If we add this "import all", we should remove all the other import statements

**Review by mimaison — APPROVED** [review]
LGTM
I've left a comment for a minor improvement

File: core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala (hunk 1 above)
Comment (viktorsomogyi) [reply core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala:30]: Good point, let me quickly correct this.

**Comment by omkreddy** [comment]
retest this please

**Review by omkreddy — APPROVED** [review]
@viktorsomogyi Thanks for the PR. LGTM.

**Pull request closed without merging**